src/Npgsql/Internal/Resolvers/JsonTypeInfoResolvers.cs

```diff
@@ -14,19 +14,13 @@ class JsonTypeInfoResolver : IPgTypeInfoResolver
     public JsonTypeInfoResolver(JsonSerializerOptions? serializerOptions = null)
         => AddTypeInfos(Mappings, serializerOptions);
 
-    [UnconditionalSuppressMessage("Trimming", "IL2026", Justification = "Only used to request rooted and statically known types (JsonDocument,JsonElement etc).")]
-    [UnconditionalSuppressMessage("Aot", "IL3050", Justification = "Only used to request rooted and statically known types  (JsonDocument,JsonElement etc).")]
     static void AddTypeInfos(TypeInfoMappingCollection mappings, JsonSerializerOptions? serializerOptions = null)
     {
-#if NET7_0_OR_GREATER
-        serializerOptions ??= JsonSerializerOptions.Default;
-#else
         if (serializerOptions is null)
         {
             serializerOptions = new JsonSerializerOptions();
-            serializerOptions.TypeInfoResolver = new DefaultJsonTypeInfoResolver();
+            serializerOptions.TypeInfoResolver = new BasicJsonTypeInfoResolver();
```

Comment:
I was actually experimenting by making a func returning `serializerOptions` and passing that below, that also helped (though locally I can't really see any sort of difference).

Reply:
Right that would help but what we had was a bit of a hack. 

It just wasn't clear to me before how to actually instantiate a `JsonTypeInfo<T>` over some converter. Overall this should provide much better perf if users don't already pass in options.